tests/returntypes/test_jax_qnode_new.py

```diff
@@ -0,0 +1,2209 @@
+        dev = qml.device(dev_name, wires=2, shots=shots)
+
+        if diff_method == "backprop":
+            pytest.skip("Backpropagation is unsupported if shots > 0.")
+
+        @qml.qnode(dev, interface="jax", diff_method="parameter-shift")
+        def circuit():
+            return qml.probs(wires=0)
+
+        assert jnp.allclose(circuit(), jnp.array([1.0, 0.0]))
+
+    @pytest.mark.xfail(
+        reason="Non-trainable parameters are not being correctly unwrapped by the interface"
+    )
+    def test_gradient_subset(self, dev_name, diff_method, mode, tol):
+        """Test derivative calculation of a scalar valued QNode with respect
+        to a subset of arguments"""
+        a = jnp.array(0.1)
+        b = jnp.array(0.2)
+
+        dev = qml.device(dev_name, wires=1)
+
+        @jax.jit
+        @qnode(dev, diff_method=diff_method, interface="jax", mode=mode)
+        def circuit(a, b):
+            qml.RY(a, wires=0)
+            qml.RX(b, wires=0)
+            qml.RZ(c, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        res = jax.grad(circuit, argnums=[0, 1])(a, b, 0.0)
+
+        expected_res = np.cos(a) * np.cos(b)
+        assert np.allclose(res, expected_res, atol=tol, rtol=0)
+
+        expected_g = [-np.sin(a) * np.cos(b), -np.cos(a) * np.sin(b)]
+        assert np.allclose(g, expected_g, atol=tol, rtol=0)
+
+    def test_gradient_scalar_cost_vector_valued_qnode(self, dev_name, diff_method, mode, tol):
+        """Test derivative calculation of a scalar valued cost function that
+        uses the output of a vector-valued QNode"""
+        dev = qml.device(dev_name, wires=2)
+
+        if diff_method == "adjoint":
+            pytest.xfail(reason="The adjoint method is not using host-callback currently")
+
+        @jax.jit
+        @qnode(dev, diff_method=diff_method, interface="jax", mode=mode)
+        def circuit(x, y):
+            qml.RX(x, wires=[0])
+            qml.RY(y, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.probs(wires=[1])
+
+        def cost(x, y, idx):
+            res = circuit(x, y)
+            return res[idx]
+
+        x = jnp.array(1.0)
+        y = jnp.array(2.0)
+        expected_g = (
+            np.array([-np.sin(x) * np.cos(y) / 2, np.cos(y) * np.sin(x) / 2]),
+            np.array([-np.cos(x) * np.sin(y) / 2, np.cos(x) * np.sin(y) / 2]),
+        )
+
+        idx = 0
+        g0 = jax.grad(cost, argnums=0)(x, y, idx)
+        g1 = jax.grad(cost, argnums=1)(x, y, idx)
+        assert np.allclose(g0, expected_g[0][idx], atol=tol, rtol=0)
+        assert np.allclose(g1, expected_g[1][idx], atol=tol, rtol=0)
+
+        idx = 1
+        g0 = jax.grad(cost, argnums=0)(x, y, idx)
+        g1 = jax.grad(cost, argnums=1)(x, y, idx)
+
+        assert np.allclose(g0, expected_g[0][idx], atol=tol, rtol=0)
+        assert np.allclose(g1, expected_g[1][idx], atol=tol, rtol=0)
+
+
+# TODO: add default mixed and lightning
```

Comment:
Curious: are the tests to follow failing with those devices right now? Even with `default.mixed`?

Reply:
Default mixed here is a mistake, I will remove it.